### ISSUE TITLE
NaN after the first iteration of training.

### ISSUE THREAD

**Issue by codeslake** [sniklaus/pytorch-liteflownet#15]
Hi, thank you for your fabulous work.
I've been trying to implement a training code for the pytorch version of liteflownet.

Everything seems to be working fine except `def foward()` defined in `Regularization` class.
(It was `correlation` module that I suspected the most but it is working fine.)
I get "nan" loss value after one iteration. It seems that gradient become nan at the `foward` block.

If I change line 279 to `tensorDifference = (tensorFirst - tensorSecond).pow(2.0).sum(1, True).sqrt()
` loss never becomes "nan".

I am kind of having hard time with the code because I've never used pytorch (I am getting it though).
Do you have any idea about this?
Thank you

**Comment by sniklaus** [comment]
I am afraid that your proposed version does no longer reflect a occlusion probability map. As such, it would in my opinion no longer be a truthful recreation of LiteFlowNet.

A more seminal way would be, again in my opinion, to disable backprop through this probability map. I have just made the necessary change in 65792d41 and am closing this issue for now. Please let me know in case the training is still naning out with the new version.

**Comment by codeslake** [comment]
Thanks for your reply.

I just wanted to let you know the part where gradients turn to be NaN.
Is it OK to detaching the part? And can you explain why it makes the gradient NaN? I don't see any reason it to be detached.

Thank you.

**Comment by sniklaus** [comment]
Ah, I see, thank you for helping me locate the source of the nans!

I think it is okay since it is just used as a filter, so backpropping through it actually may defeat the purpose of what this component was designed to do. I have to admit that I did not read that section in the paper thoroughly though. I am thus kindly pinging @twhui to get his expert opinion on this.

**Comment by twhui** [comment]
I don't have this NaN problem on Caffe.

**Comment by sniklaus** [comment]
Thank you for your input Tak-Wai! Just for clarification, do you backpropagate through the brightness error (footnote 4)?

**Comment by twhui** [comment]
Yes :)

**Comment by sniklaus** [comment]
I was hoping you didn't. 😄 

So the fix in 65792d4 is preventing the training from naning out but it is not truthful to the original implementation. I am afraid though, that I have no immediate idea on how to address this.
